### DISCUSSION TITLE
How to assert/expect or mock when external app is opened.

### DISCUSSION THREAD

**Discussion by jagadeeshpurplle** [appium/appium#17748] [Q&A]
Hi Appium Team,

I'm using `appium-flutter-driver` & `webdriverIO` to automate the flutter mobile app.

I have a use case in my application where clicking on

- `Mail us` button opens `Gmail app` with subject, body
- `Call us` button opens `Dailer app` with phone number

I want assert/verify that gmail/phone app is opened.

- Is it possible to verify that gmail/phone app package name at that moment?
- Or any other way to assert this kind of use cases?
- Or verifying the subject, content in gmail compose screen is also fine
- Or can we mock that app opening and return true always as like we do the intent mocking in `espresso`.!

Here is the video of app application action:

https://user-images.githubusercontent.com/25878053/200753394-a175cfb9-c931-40b9-a26f-1688a265db02.mov

Can anybody help me?

@KazuCocoa

**Comment by KazuCocoa** [comment]
http://appium.io/docs/en/commands/device/activity/current-activity/ in native app context returns the current active package name, it may help.
Asking questions in https://discuss.appium.io/ may also help

**Reply by jagadeeshpurplle** [reply]
Hi @KazuCocoa  thanks for responding. 

can we mock that app opening and return true always as like we do the `intent mocking in espresso`?

**Reply by KazuCocoa** [reply]
Do you mean stubbing in https://developer.android.com/training/testing/espresso/intents?
Potentially appium espresso server can if someone implements it, but not in UIA2 driver.

**Reply by jagadeeshpurplle** [reply]
yeah i mean stubbing only.!

However, I want to interact with native gmail app when it got opened from flutter app(`as shown in the vide above.!`)

How to perform click actions in gmail app? 
I see here https://github.com/appium-userland/appium-flutter-driver that 
```
await driver.switchContext('NATIVE_APP');
await (await driver.$('~fab')).click();
```
, what is `~fab` means here? and how to find elements using ID, text, class in this case?

After opening the native gmail app, 
- Here in this popup, I want to click on `OK`.
<img width="221" alt="Screenshot 2022-11-09 at 3 25 12 PM" src="https://user-images.githubusercontent.com/25878053/200799336-a5bc938e-966a-4619-b40f-bfa137d801bb.png">
- After that i want to click on `3 dots button` and click on `Discard` (As shown below)
<img width="183" alt="Screenshot 2022-11-09 at 3 27 30 PM" src="https://user-images.githubusercontent.com/25878053/200799906-b17a8973-f051-4f43-9ee2-923f8bac95b9.png">

How can we achieve this in `appium-flutter-driver` & `WebdriverIO` project?

@KazuCocoa
